app/controllers/application_controller.rb

```diff
@@ -40,4 +41,11 @@ def check_entitled(entitlement)
 
     required_entitlements.map { |e| entitlement.send(e) }.all?
   end
+
+  def pundit_user
+    UserContext.new(
+      Insights::API::Common::Request.current!,
+      params
```

Comment:
@hsong-rh Do you need the params in your User context objects, In catalog we have one corner case where we need to check for the params for portfolio to copy to. If possible we should avoid params, since the policy scopes gets passed in the actual object. This also has problems once you go to user capabilities because those params won't be there when you are listing an object.

Reply:
Yes, I know `params` will not available for user capability. But Approval also has a corner case when we validate to create `action`,  we need to query `random_access_key` from its request defined in `params`.